### PR TITLE
Test vectors

### PR DESCRIPTION
**Pull request by martinthomson**
This takes @rpaulo's example and expands on it, hopefully without destroying it.

In summary:

* the output is condensed a little, which keeps the giant client Initial somewhat more in check
* added formula and variable names consistent with the rest of the spec for context
* added packet protection

The known bug here is that the ClientHello doesn't include transport parameters.  The ServerHello should be OK.  And I think that the ACK is correct also.  I did try to get the connection IDs and packet numbers correct.

I would appreciate someone checking my work.  I don't have a convenient method of testing this.

Closes #1573.

### CODE REVIEW

**Comment by Ralith** [comment]
I've independently replicated the key derivation and the server initial encryption given here.

**Comment by martinthomson** [comment]
Another question: would checking in the script I used to produce these be helpful for anyone?

**Comment by rpaulo** [comment]
Nice!  I think the script would be helpful!

File: initial-protection.js [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+// This script performs simple encryption and decryption for Initial packets.
+// It's crude, but it should be sufficient to generate examples.
+
+
+'use strict';
+var buffer = require('buffer');
+var crypto = require('crypto');
+var assert = require('assert');
+
+var INITIAL_SECRET = Buffer.from('ef4fb0abb47470c41befcf8031334fae485e09a0', 'hex');
```
Comment (ghedo) [inline initial-protection.js]: INITIAL_SALT?

**Comment by kazuho** [comment]
I've added the test vector to the unit test of quicly, so that I can continue confirming the values as the draft evolves. And of course, the current test vector decrypts successfully.

**Comment by martinthomson** [comment]
Thanks everyone for checking these values.  At @marten-seemann's request, I've tweaked the packet numbers.  That means new values, but I'll assume that these are OK.

**Comment by kazuho** [comment]
Confirmed that the updated version correctly decrypts and gives PN=2.

**Comment by martinthomson** [comment]
Thanks @kazuho, that's enough for me :)

**Comment by marten-seemann** [comment]
Thanks @martinthomson! I updated my test vectors in quic-go, and I can confirm that the new values work.

**Pull request merged**